js/app.js

```diff
@@ -97,7 +171,15 @@ function createDefaultSnippetPreview() {
  * @constructor
  */
 YoastSEO.App = function( args ) {
-	this.config = this.extendConfig( args );
+	if ( !isObject( args ) ) {
+		args = {};
+	}
+	defaultsDeep( args, defaults );
+
+	this.config = args;
+
+	checkArguments( this.config );
```

Comment:
We should move this above the `this.config = args` line. We should check inputs before we assign them to our instance properties.